src/lib/Dungeon.js

```diff
@@ -282,137 +325,279 @@ class AutomationDungeon
                         return;
                     }
 
-                    DungeonRunner.map.moveToTile(chestLocation);
+                    this.__internal__moveToCell(chestLocation);
+                    DungeonRunner.openChest();
+                    return;
                 }
                 else if (DungeonRunner.map.playerPosition().floor < (DungeonRunner.map.floorSizes.length - 1))
                 {
-                    // Reset current floor states before moving to the next one
-                    this.__internal__chestPositions = [];
-                    this.__internal__isCompleted = false;
-                    DungeonRunner.nextFloor();
-                }
-                else
-                {
-                    DungeonRunner.map.moveToTile(this.__internal__bossPosition);
-                }
-            }
-
-            let playerCurrentPosition = DungeonRunner.map.playerPosition();
-
-            if (DungeonRunner.map.currentTile().type() === GameConstants.DungeonTile.ladder)
-            {
-                if (Automation.Utils.LocalStorage.getValue(this.Settings.BossRush) === "true")
-                {
-                    this.__internal__isCompleted = true;
+                    if (visibleEndTile)
+                    {
+                        this.__internal__moveToCell(this.__internal__bossPosition);
+                        // Reset current floor states before moving to the next one
+                        this.__internal__chestPositions = [];
+                        this.__internal__bossPosition = null;
+                        // Do not call DungeonRunner.nextFloor() directly, as it has no checks
```

Comment:
That was the point, if we are here, we should always be in the state that would lead directly to this call (and avoid all those check overhead).

If it's not the case, we should do it manually, since resetting boss chests and boss position would make us in an undefined state

Reply:
That's fine as-is for now, if any perf overhead is found their, will think about it at that moment